src/index.ts

```diff
@@ -158,6 +158,10 @@ async function resolveTarStreamOnFinish(
 	pack: tar.Pack,
 	dockerfile?: string,
 ): Promise<void> {
+	const noBuildFilesFoundErrorString = `Could not find a build file, supported build types: ${resolvers
+		.map(r => r.name)
+		.join(', ')}`;
```

Comment:
A couple of suggestions:
* "build file" and "build types" are definitely clearer than "resolution could not be performed", but I think some people might still get confused. Wording suggestion: `One of the following files is required and could not be found: ${...}`
* Assembling this error message involves running some code (`resolvers.map().join()`), so it could be moved inside the `if (satisfied.length === 0) { }` block (where it is needed). And that block could even be slightly refactored along the lines of:

```
if (satisfied.length === 0) {
    let msg: string;
    if (dockerfile) {
        msg = `File not found: "${dockerfile}"`;
    } else {
        msg = `One of the following files is required and could not be found: ${
            resolvers.map(r => r.name).join(', ')}`;
    }
    pack.emit('error', new Error(msg));
...
```

Reply:
The problem is these aren't files, and instead are "friendly names", i.e `Standard Dockerfile`, `Arhcitecture Dockerfile`, `Template Dockerfile`

Reply:
This is what the error looks like right now (wired into balena-cli)
```
There was en error processing build files for service: something
	Error: Could not find a build file, supported build types: Standard Dockerfile, Dockerfile.template, Architecture-specific Dockerfile, NodeJS
```

Reply:
I see. It's definitely an improvement to what we had before, but "a build file" + "build types" is still a bit vague in my opinion (from the point of view of a support agent, and someone who answers CLI issues created by users who are trying the featured Etcher projects and using the CLI for the first time...). Then how about not printing the resolver names, and instead printing a plain constant string:

`"Error: Could not find a Dockerfile or package.json file"`

This constant message does not mention the template and arch-specific Dockerfiles, but they are sort of covered by "a Dockerfile". Then also, when we bump resin-multibuild / resin-bundle-resolve in the CLI, we could perhaps also add (in CLI code) another hint: "Did you 'cd' to the right folder?" (we can test whether they provided the `-s` option).

Btw, I also thought in the past that the CLI could have a bit of code that checked for docker-compose / Dockerfile / Dockerfile.template / Dockerfile.arch / package.json in the root of the context / project before even creating the tar stream, as this would cover the "category" of errors where users issue the "balena push/build/deploy" command in the wrong folder.

Reply:
Yeah ok I see what you're saying, and it's certainly a more precise error message, will change :+1:

Reply:
I think it's probably worth not adding the nodejs entry now that I'm thinking about it. It needs some serious work to be usable, so we don't really want people using it as the UX is absoutely terrible.